util/dvsim/Scheduler.py

```diff
@@ -76,6 +82,10 @@ def __init__(self, items):
         # per-target.
         self.item_to_status = {}
 
+        # The chosen launcher class. This allows us to access launcher
+        # variant-specific settings such as max parallel jobs & poll rate.
+        self.launcher_cls = get_launcher_cls()
```

Comment:
This looks a bit odd to me. Could we pass the chosen launcher class in to the Scheduler constructor? That would move some of the "system construction wiring" logic further up the tree, which I think makes things a bit cleaner.

Reply:
Done.